### PR TITLE
Saas 17 stock usability imp hsh

### PR DESCRIPTION
**Pull request by hsh-odoo**
Description of the issue/feature this PR addresses:
- Task : https://www.odoo.com/web#id=35918&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
- Pad : https://pad.odoo.com/p/r.c7acd217b5c467e2b2c5a8e9b7674049

Current behavior before PR:
- Usability improvement in stock

Desired behavior after PR is merged:
- Usability improvement is stock

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/test_main_flows/static/src/js/tour.js [hunk 1]
```diff
@@ -407,14 +407,16 @@ tour.register('main_flow_tour', {
 }, {
 // Run the schedulers
     edition: "enterprise",
-    trigger: ".o_menu_sections a:contains('Schedulers')",
-    content: _t("Go to Schedulers"),
+    trigger: ".o_menu_sections a:contains('Operations')",
+    content: _t("Go to Scheduler"),
```
Comment (jorenvo) [inline odoo/addons/test_main_flows/static/src/js/tour.js]: This doesn't look right. The tour should now click on 'Schedulers' so it expands and then on 'Run Scheduler'.

File: odoo/addons/test_main_flows/static/src/js/tour.js [hunk 2]
```diff
@@ -407,14 +407,16 @@ tour.register('main_flow_tour', {
...
     position: "bottom"
 }, {
-    trigger: ".o_menu_sections a:has(span:contains('Run Schedulers')), .oe_secondary_submenu .oe_menu_text:contains('Run Schedulers'):first",
-    content: _t("Run the schedulers"),
+    edition: "enterprise",
+    trigger: ".o_menu_sections a:has(span:contains('Run Scheduler')), .oe_secondary_submenu .oe_menu_text:contains('Run Scheduler'):first",
+    content: _t("Run the scheduler"),
```
Comment (jorenvo) [inline odoo/addons/test_main_flows/static/src/js/tour.js]: This should work without modification I think. The text on the button in the wizard still says 'Run Scheduler**s**'. Also why did you add `edition: "enterprise",`?

File: odoo/addons/test_main_flows/static/src/js/tour.js [hunk 3]
```diff
@@ -407,14 +407,16 @@ tour.register('main_flow_tour', {
...
     position: "bottom"
 }, {
+    edition: "enterprise",
```
Comment (jorenvo) [inline odoo/addons/test_main_flows/static/src/js/tour.js]: Same here, why is this necessary now and not before?

File: addons/stock/models/stock_move_line.py [hunk 4]
```diff
@@ -42,10 +43,15 @@ class StockMoveLine(models.Model):
...
-    state = fields.Selection(related='move_id.state')
+    state = fields.Selection(related='move_id.state', store=True)
     consume_line_ids = fields.Many2many('stock.move.line', 'stock_move_line_consume_rel', 'consume_line_id', 'produce_line_id', help="Technical link to see who consumed what. ")
     produce_line_ids = fields.Many2many('stock.move.line', 'stock_move_line_consume_rel', 'produce_line_id', 'consume_line_id', help="Technical link to see which line was produced with this. ")
 
+    @api.multi
+    def compute_uom_id(self):
+        for line in self:
+            line.uom_reserved_id = line.product_uom_id
+
```
Comment (jorenvo) [inline addons/stock/models/stock_move_line.py]: Same comment as for `uom_reserved_id` on `stock.quant`. Don't think we need this here.

File: addons/stock/models/stock_quant.py [hunk 5]
```diff
@@ -41,9 +45,27 @@ class StockQuant(models.Model):
         default=0.0,
         help='Quantity of reserved products in this quant, in the default unit of measure of the product',
         readonly=True, required=True)
+    uom_reserved_id = fields.Many2one('product.uom', compute="compute_uom_id")
     in_date = fields.Datetime('Incoming Date', readonly=True)
 
     @api.multi
+    def compute_uom_id(self):
+        for quant in self:
+            quant.uom_reserved_id = quant.product_uom_id
```
[the remark this answers is not in the document]
Comment (jir-odoo) [reply addons/stock/models/stock_quant.py]: We have added this compute field to just show uom two times in form view and it is readonly and it will not effact anywhere.
